RPT_Application/src/main/java/com/harman/user_database/EmployeeService.java

```diff
@@ -0,0 +1,68 @@
+package com.harman.user_database;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EmployeeService {
+
+	@Autowired
+	private EmployeeRepository employeeRepository;
+
+	@Autowired
+	private BCryptPasswordEncoder passwordEncoder;
+
+	private static final Logger LOGGER = Logger.getLogger(EmployeeService.class.getName());
+
+	private EmployeeService() {
+	}
+
+	public List<Employee> findAll() {
+		return employeeRepository.findAll();
+	}
+
+	public int count() {
+		return (int) employeeRepository.count();
+	}
+
+	public Employee findByUsername(String username) {
+		return employeeRepository.findByUsername(username);
+	}
+
+	public Employee findById(String id) {
+		return employeeRepository.findById(id).orElse(null);
+	}
+
+	public boolean delete(Employee employee) { //false - employee wasn't deleted, true - employee was deleted
+		if (employee == null) {
+			LOGGER.log(Level.SEVERE, "Employee is null. Check why do you delete null Employee.");
+			return false;
+		}
+		if (employee.getAuthority() == "ROLE_ADMIN") {
+			LOGGER.log(Level.SEVERE, "Employee is admin. No one is allowed to remove admin.");
+			return false;
+		}
+		employeeRepository.delete(employee);
+		return true;
+	}
+
+	public void save(Employee employee) {
+		if (employee == null) {
+			LOGGER.log(Level.SEVERE, "Employee is null. And that's bad. Check why do you save null Employee.");
+			return;
+		}
+		employeeRepository.save(employee);
```

Comment:
Don't we need the saved employee or at least id to update the entity?

Reply:
heck, i don't know :c

Reply:
i believe that Employee entity is being updated automatically when .save is called